src/Rule/Each.php

```diff
@@ -57,6 +57,7 @@ public function validate($value)
 
         $result = true;
         foreach ($value as $index => $innerValue) {
+            $innerValue['array_index'] = $index;
```

Comment:
That was one of my doubts as well. I only added the underscore because the output reads better when you do it, but it could be anything.